packages/plugin-core/src/_extension.ts

```diff
@@ -126,6 +128,87 @@ class ExtensionUtils {
     }
   };
 
+  static async runMigrationsIfNecessary({
```

Comment:
Should we move some of these to the new `StartupUtils`? @hikchoi

Reply:
that makes sense.

Reply:
moved